File: leetcode/0219.Contains-Duplicate-II/ContainsDuplicateII_test.py

from ContainsDuplicateII import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
    ([1,2,3,1], 3, True),
    ([1,0,1,1], 1, True),
    ([1,2,3,1,2,3], 2, False),
    ([99, 99], 2, True),
    ([1,2,1], 0, False)
]

@pytest.mark.parametrize(
    "nums, k, ans", 
    deepcopy(TEST_SUITS)
)
def test(nums, k, ans):
    assert SOL.containsNearbyDuplicate(nums, k) == ans
    

File: leetcode/0219.Contains-Duplicate-II/ContainsDuplicateII_test.go
package leetcode

import (
	"reflect"
	"testing"
)

func Test(t *testing.T) {
	testSuits := []struct {
		name string
		nums []int
		k    int
		ans  bool
	}{
		{"Example 1", []int{1, 2, 3, 1}, 3, true},
		{"Example 2", []int{1, 0, 1, 1}, 1, true},
		{"Example 3", []int{1, 2, 3, 1, 2, 3}, 2, false},
		{"Example 4", []int{99, 99}, 2, true},
		{"Example 5", []int{1, 2, 1}, 0, false},
	}

	for _, test := range testSuits {
		got := containsNearbyDuplicate(test.nums, test.k)

		if !reflect.DeepEqual(got, test.ans) {
			t.Errorf("[%v] expect '%v', got '%v' ", test.name, test.ans, got)
		}
	}
}
